components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/CarbonEventBroker.java

```diff
@@ -61,63 +61,56 @@ public void init() throws EventBrokerConfigurationException {
         // notify method with the given subscription details.
         // this way delivery manager can use the persisted repository of the subscription manager
         // to notify subscriptions if wanted.
-        this.delivaryManager.setNotificationManager(this.notificationManager);
+        this.deliveryManager.setNotificationManager(this.notificationManager);
         // re subscribe the already existing subscriptions.
         // TODO: do the validations eg. expiraty time
         loadExistingSubscriptions();
     }
 
+    /**
+     * Loads existing subscriptions. Delivery manager is subscribed with non expired subscriptions
+     *
+     * @throws EventBrokerConfigurationException
+     */
     private void loadExistingSubscriptions() throws EventBrokerConfigurationException {
         try {
             Calendar calendar = Calendar.getInstance();
-            for (Subscription subscription : this.subscriptionManager.getAllSubscriptions()){
-                if ((subscription.getExpires() == null) || (calendar.before(subscription.getExpires()))){
+            for (Subscription subscription : this.subscriptionManager.getAllSubscriptions()) {
+                if ((subscription.getExpires() == null) ||
```

Comment:
Could we swap between the expression ? to avoid NullPointer during the AutoBoxing